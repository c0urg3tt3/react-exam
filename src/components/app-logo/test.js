import React from 'react'
import { shallow } from 'enzyme'

import AppLogo from './component'

describe('[component] AppLogo', () => {
  it('should render component', () => {
    const wrapper = shallow(<AppLogo/>)

    expect(wrapper.exists()).toEqual(true)
    expect(wrapper.is('img')).toEqual(true)
    expect(wrapper.hasClass('app-logo')).toEqual(true)
    expect(wrapper.hasClass('rotate')).toEqual(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('animation can be disabled', () => {
    const wrapper = shallow(<AppLogo/>)

    expect(wrapper.hasClass('rotate')).toEqual(true)
    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps({ isAnimate: false })

    expect(wrapper.hasClass('rotate')).toEqual(false)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should replace logo src by imageURL prop', () => {
    const wrapper = shallow(<AppLogo/>)
    const src = wrapper.prop('src')
    const imageURL = "http://www.dark-vador.net/"

    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps({ imageURL })

    expect(wrapper.prop('src')).not.toEqual(src)
    expect(wrapper.prop('src')).toEqual(imageURL)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
