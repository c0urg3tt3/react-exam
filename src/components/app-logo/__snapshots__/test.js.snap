// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] AppLogo animation can be disabled 1`] = `"<img alt=\\"app logo\\" src=\\"logo_jedi_order.svg\\" class=\\"app-logo rotate\\"/>"`;

exports[`[component] AppLogo animation can be disabled 2`] = `"<img alt=\\"app logo\\" src=\\"logo_jedi_order.svg\\" class=\\"app-logo\\"/>"`;

exports[`[component] AppLogo should render component 1`] = `"<img alt=\\"app logo\\" src=\\"logo_jedi_order.svg\\" class=\\"app-logo rotate\\"/>"`;

exports[`[component] AppLogo should replace logo src by imageURL prop 1`] = `"<img alt=\\"app logo\\" src=\\"logo_jedi_order.svg\\" class=\\"app-logo rotate\\"/>"`;

exports[`[component] AppLogo should replace logo src by imageURL prop 2`] = `"<img alt=\\"app logo\\" src=\\"http://www.dark-vador.net/\\" class=\\"app-logo rotate\\"/>"`;
